internal/core/src/common/Array.h

```diff
@@ -0,0 +1,457 @@
+                }
+                break;
+            }
+            default: {
+                PanicInfo("unsupported data type");
+            }
+        }
+        return data_array;
+    }
+
+    int
+    length() const {
+        return length_;
+    }
+
+    int
+    size() const {
+        return size_;
+    }
+
+    DataType
+    get_element_type() const {
+        return element_type_;
+    }
+
+    const void*
+    data() const {
+        return data_;
+    }
+
+    bool
+    is_same_array(const proto::plan::Array& arr2) const {
+        if (arr2.array_size() != length_) {
+            return false;
+        }
+        if (!arr2.same_type()) {
+            return false;
+        }
+        switch (element_type_) {
+            case DataType::BOOL: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<bool>(i);
+                    if (val != arr2.array(i).bool_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            case DataType::INT8:
+            case DataType::INT16:
+            case DataType::INT32: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<int>(i);
+                    if (val != arr2.array(i).int64_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            case DataType::INT64: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<int64_t>(i);
+                    if (val != arr2.array(i).int64_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            case DataType::FLOAT: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<float>(i);
+                    if (val != arr2.array(i).float_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            case DataType::DOUBLE: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<double>(i);
+                    if (val != arr2.array(i).float_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            case DataType::VARCHAR:
+            case DataType::STRING: {
+                for (int i = 0; i < length_; i++) {
+                    auto val = get_data<std::string>(i);
+                    if (val != arr2.array(i).string_val()) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            default:
+                return false;
+        }
+    }
+
+ private:
+    char* data_{nullptr};
```

Comment:
this data_ must delete it in decontruct function manually, unless will cause memory leak

Reply:
ok，added it.